nodes/obstacle_visualizer.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/python3
+
+from typing import List, Dict
+
+import rospy
+
+from std_msgs.msg import Header
+from geometry_msgs.msg import Pose
+from tf.transformations import quaternion_from_euler
+from geometry_msgs.msg import Point, Quaternion, Vector3
+from visualization_msgs.msg import Marker, MarkerArray
+from std_msgs.msg import ColorRGBA, Header
+from SDFGenerator import SDFGenerator
+
+from gazebo_msgs.srv import SpawnModel, SpawnModelRequest
+
+
+class ObstaclesVisualizer:
+    def __init__(self, name: str) -> None:
+        self._node_name = name
+        rospy.init_node(self._node_name, anonymous=False)
+
+        obstacle_idx = 1
+
+        if not rospy.has_param("~obstacle1"):
+            rospy.logerr(
+                f"[{rospy.get_name()}] Param obstacle1 not found. No collision avoidance with an external obstacle computed!"
+            )        
+
+        self._obstacles_infos = {}
+        self._obstacles = []
+        while rospy.has_param("~obstacle" + str(obstacle_idx)):
+            obstacle_name = "~obstacle" + str(obstacle_idx)
+            self._obstacles_infos.update(
+                {obstacle_name: rospy.get_param(obstacle_name)}
+            )
+            obstacle_idx += 1
+
+        sdf_generator = SDFGenerator()
+        self._spawn_model_requests = []
+        self._markers = MarkerArray()
+        header = Header(frame_id="world", stamp=rospy.Time.now())
+        for iter, (key, obstacle) in enumerate(self._obstacles_infos.items()):
+            
+            pose=self._parse_poses(obstacle)
+ 
+            sp_req = SpawnModelRequest(
+                model_name = key[1:],
```

Comment:
```suggestion
           obstacle_name = key.lstrip("~")
            sp_req = SpawnModelRequest(
                model_name = obstacle_name,
```
This way the code will be more scallable